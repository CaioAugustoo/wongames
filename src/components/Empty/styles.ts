import styled from 'styled-components'

export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

export const Image = styled.img``

export const Title = styled.h2``

export const Description = styled.p``
