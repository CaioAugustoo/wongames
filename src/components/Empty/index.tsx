import Button from 'components/Button'
import Link from 'next/link'
import * as S from './styles'

export type EmptyProps = {
  title: string
  description: string
  hasLink?: boolean
}

const Empty = ({ title, description, hasLink }: EmptyProps) => (
  <S.Wrapper>
    <S.Image
      role="img"
      src="/img/empty.svg"
      alt="Um gamer jogando vídeo-game num sofá"
    />

    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    {hasLink && (
      <Link href="/" passHref>
        <Button as="a">Voltar para a loja</Button>
      </Link>
    )}
  </S.Wrapper>
)

export default Empty
