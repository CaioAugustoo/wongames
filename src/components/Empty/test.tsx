import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Empty from '.'

const props = {
  title: 'Your wishlist is empty',
  description: 'A simple description'
}

describe('<Empty />', () => {
  it('should render correctly', () => {
    renderWithTheme(<Empty {...props} hasLink />)

    expect(
      screen.getByRole('img', {
        name: /Um gamer jogando vídeo-game num sofá/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Your wishlist is empty/i })
    ).toBeInTheDocument()
    expect(screen.getByText(/A simple description/i)).toBeInTheDocument()
    expect(
      screen.getByRole('link', { name: /Voltar para a loja/i })
    ).toHaveAttribute('href', '/')
  })

  it('should not render link when not passed', () => {
    renderWithTheme(<Empty {...props} />)

    expect(
      screen.queryByRole('link', { name: /Voltar para a loja/i })
    ).not.toBeInTheDocument()
  })
})
