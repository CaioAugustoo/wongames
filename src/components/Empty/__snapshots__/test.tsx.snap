// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render correctly 1`] = `
.c4 {
  background: #47c8ff;
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c4:hover {
  background: #2499ae;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  max-width: 100%;
}

.c2 {
  color: #47C8FF;
  font-size: 2.8rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

<body>
  <div>
    <main
      class="c0"
    >
      <img
        alt="Um gamer jogando vídeo-game num sofá"
        class="c1"
        role="img"
        src="/img/empty.svg"
      />
      <h2
        class="c2"
      >
        Your wishlist is empty
      </h2>
      <p
        class="c3"
      >
        A simple description
      </p>
      <a
        class="c4"
        href="/"
      >
        <span>
          Voltar para a loja
        </span>
      </a>
    </main>
  </div>
</body>
`;
